consensus/src/dag/dag_handler.rs

```diff
@@ -91,51 +106,95 @@ impl NetworkHandler {
             },
         );
 
+        let mut futures = FuturesUnordered::new();
+        // A separate executor to ensure the message verification sender (above) and receiver (below) are
+        // not blocking each other.
+        let executor = BoundedExecutor::new(8, Handle::current());
```

Comment:
probably want to make this configurable, and may need to increase the concurrency numbers, for verification as well

Reply:
yes, added a TODO, I need to still bring in the verification runtime based executor that Sital added.